apps/studio/components/interfaces/Settings/General/ComplianceConfig/ProjectComplianceMode.tsx

```diff
@@ -0,0 +1,126 @@
+
+import { useParams } from 'common'
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import { DocsButton } from 'components/ui/DocsButton'
+import { FormHeader } from 'components/ui/Forms/FormHeader'
+import { FormPanel } from 'components/ui/Forms/FormPanel'
+import { FormSection, FormSectionContent, FormSectionLabel } from 'components/ui/Forms/FormSection'
+import { useProjectSettingsV2Query } from 'data/config/project-settings-v2-query'
+import { useComplianceConfigUpdateMutation } from 'data/config/project-compliance-config-mutation'
+import { useCheckPermissions } from 'hooks/misc/useCheckPermissions'
+import { subscriptionHasHipaaAddon } from 'components/interfaces/Billing/Subscription/Subscription.utils'
+import { useOrgSubscriptionQuery } from 'data/subscriptions/org-subscription-query'
+import { useSelectedOrganization } from 'hooks/misc/useSelectedOrganization'
+
+import { Switch, TooltipContent_Shadcn_, TooltipTrigger_Shadcn_, Tooltip_Shadcn_ } from 'ui'
+
+const ComplianceConfig = () => {
+  const { ref } = useParams()
+  const [isSensitive, setIsSensitive] = useState(false)
+  const selectedOrganization = useSelectedOrganization()
+  const { data: subscription } = useOrgSubscriptionQuery({ orgSlug: selectedOrganization?.slug })
+  const hasHipaaAddon = subscriptionHasHipaaAddon(subscription)
+  const {
+    data: settings,
+    isLoading,
+    isSuccess,
+  } = useProjectSettingsV2Query({ projectRef: ref }, { enabled: hasHipaaAddon })
+
+  const { mutate: updateComplianceConfig, isLoading: isSubmitting } =
+    useComplianceConfigUpdateMutation({
+      onSuccess: () => {
+        toast.success('Successfully updated project compliance configuration')
+      },
+      onError: (error) => {
+        setIsSensitive(initialIsSensitive)
+        toast.error(`Failed to update project compliance configuration: ${error.message}`)
+      },
+    })
+
+  const { project } = useProjectContext()
+  const canUpdateComplianceConfig = useCheckPermissions(PermissionAction.UPDATE, 'projects', {
+    resource: {
+      project_id: project?.id,
+    },
+  })
+  const initialIsSensitive = settings?.is_sensitive || false
+
+  useEffect(() => {
+    if (!isLoading) {
+      setIsSensitive(initialIsSensitive)
+    }
+  }, [isLoading])
+
+  const toggleIsSensitive = async () => {
+    if (!ref) return console.error('Project ref is required')
+    setIsSensitive(!isSensitive)
+    updateComplianceConfig({ projectRef: ref, isSensitive: !isSensitive })
+  }
+
+  // this is only setable on compliance orgs, currently that means HIPAA orgs
+  if (!hasHipaaAddon) {
+    return
+  }
+
+  return (
+    <div id="compliance-configuration">
+      <div className="flex items-center justify-between mb-6">
+        <FormHeader className="mb-0" title="High Compliance Configuration" description="" />
+        <DocsButton href="https://supabase.com/docs/guides/deployment/shared-responsibility-model#managing-healthcare-data" />
```

Comment:
does the docs need updating? i don't see any related content from this URL about this switch in the dashboard